nodewatcher/modules/devices/ubnt/edgerouter.py

```diff
@@ -0,0 +1,54 @@
+from django.utils.translation import ugettext_lazy as _
+
+from nodewatcher.core.generator.cgm import base as cgm_base, protocols as cgm_protocols, devices as cgm_devices
+
+
+class UBNTEdgerouterX(cgm_devices.DeviceBase):
+    """
+    UBNT EdgerouterX device descriptor.
+    """
+
+    identifier = 'ub-edgerouter-x'
+    name = "Edgerouter X"
+    manufacturer = "Ubiquiti"
+    url = 'http://www.ubnt.com/'
+    architecture = 'ramips_mt7621'
+    switches = [
+        cgm_devices.Switch(
+            'sw0', "Switch0",
+            ports=5,
+            cpu_port=6,
+            vlans=16,
+            configurable=True,
+            presets=[
+                cgm_devices.SwitchPreset('default', _("Default VLAN configuration"), vlans=[
+                    cgm_devices.SwitchVLANPreset(
+                        'lan0', "Lan0",
+                        vlan=1,
+                        ports=[0, 1, 2, 3, 4],
+                    )
+                ])
+            ]
+        )
+    ]
+    ports = [
+        cgm_devices.EthernetPort('wan0', "Wan0"),
+    ]
+    port_map = {
+        'openwrt': {
+            'sw0': cgm_devices.SwitchPortMap('switch0', vlans='eth0'),
```

Comment:
To me, this reads as follows:
* All ports (LAN and WAN) are connected via the switch (so your configuration above is wrong, you should not define the WAN port separately).
* VLANs are attached to `eth0`, so they will be `eth0.{vlan}`.
* Ports `1, 2, 3, 4` are assigned to the LAN VLAN.
* Port `0` is assigned to the WAN VLAN.
* Port `6` is connected to the CPU (so it must be in all VLANs).

I am not sure, so you should really check this on an actual device (e.g. with a default LEDE install).